File: sereal-go/mrg.pl
#!/usr/bin/env perl

use strict;
use warnings;

use Time::HiRes;
use Sereal::Encoder;
use Sereal::Decoder;

my @data;
my @files = glob("github.com/Sereal/Sereal/Go/sereal/data/*.srl");
foreach my $file (@files) {
    $/ = undef;
    open(my $fh, '<', $file) or die $!;
    my $val = <$fh>;
    push @data, $val;
}

my $start = Time::HiRes::time();

my $encoder = Sereal::Encoder->new({ dedupe_strings => 1, compress => 1 });
my $decoder = Sereal::Decoder->new();

my @events = map { $decoder->decode($_) } @data;
print $encoder->encode(\@events);

printf STDERR "The call took %.2fs to run\n", Time::HiRes::time - $start;
exit(0);

File: sereal-go/mrg.go
package main

import (
	"flag"
	"fmt"
	"github.com/Sereal/Sereal/Go/sereal"
	"io/ioutil"
	"os"
	"path/filepath"
	"sort"
	"time"
)

type srl_file struct {
	b    []byte
	name string
}

func main() {
	folder := flag.String("folder", ".", "folder to read *.srl files from")
	flag.Parse()

	files, _ := filepath.Glob(*folder + "/*.srl")
	sort.Strings(files)

	fmt.Fprintf(os.Stderr, "read data\n")

	var data []srl_file
	for _, file := range files {
		buf, ok := ioutil.ReadFile(file)
		if ok != nil {
			panic("failed to read file: " + file)
		}

		data = append(data, srl_file{buf, file})
	}

	fmt.Fprintf(os.Stderr, "merge data\n")
	t0 := time.Now()

	m := sereal.NewMerger()
	m.DedupeStrings = true
	m.KeepFlat = true

	//m.Compression = sereal.SnappyCompressor{Incremental: true}
	//m.Compression = sereal.ZlibCompressor{}

	for _, srl := range data {
		_, err := m.Append(srl.b)
		if err != nil {
			panic(err)
		}
	}

	menc, _ := m.Finish()
	fmt.Printf("%s", menc)

	t1 := time.Now()
	fmt.Fprintf(os.Stderr, "Merging took %v to run.\n", t1.Sub(t0))
}
